### PR TITLE
DOC: Add documentation to Text.set_fontfamily

### LINKED ISSUE #13880: Adding documentation for Text.fontfamily default, set_fontfamily(None)?
### Bug report

**Bug summary**

Documentation for `matplotlib.text.Text.set_fontfamily` (https://matplotlib.org/api/text_api.html#matplotlib.text.Text.set_fontfamily ) could be edited to explain:

- its value defaults to `matplotlib.rcParams['font.family']`
- constructing `Text(fontfamily=None)` sets font to `matplotlib.rcParams['font.family']`
- calling `set_fontfamily()` on an existing Text instance does the same thing
    - pycharm complains that "Expected type 'str', got 'None' instead", but the program works fine at runtime.

(Also https://matplotlib.org/api/font_manager_api.html#matplotlib.font_manager.FontProperties.set_family should be edited too.)

I confirmed the behaviors by editing my program:

- edit rcParams
- create Text (actually Annotation which subclasses Text) via `Axes.annotate`

`FontProperties._family` starts out at `rcParams['font.family']`, and `.set_family(None)` replaces argument with `rcParams['font.family']`, (both call _normalize_font_family() afterwards).

I'm not sure when this behavior was introduced.

### PR DESCRIPTION
**Pull request by Bharat123rox**
## PR Checklist

- [ ] Has Pytest style unit tests (Not needed; doc change)
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there) (Not a major new feature)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way (Not backward incompatible)

Fix #13880 by adding documentation to `Text.set_fontfamily()`

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -976,6 +976,11 @@ def set_fontfamily(self, fontname):
         strings in decreasing priority.  Each string may be either a real font
         name or a generic font class name.  If the latter, the specific font
         names will be looked up in the corresponding rcParams.
+        Defaults to `matplotlib.rcParams['font.family']`.
```
Comment (timhoffm) [inline lib/matplotlib/text.py]: Just leave out this line. `fontname` has no default; `None` has to be passed explicitly.

Also, please make the following text a separate paragraph by inserting an empty line here.

Comment (Bharat123rox) [reply lib/matplotlib/text.py]: Done

File: lib/matplotlib/text.py [hunk 2]
```diff
@@ -977,6 +977,11 @@ def set_fontfamily(self, fontname):
         name or a generic font class name.  If the latter, the specific font
         names will be looked up in the corresponding rcParams.
 
+        If a `Text` instance is constructed with `fontfamily=None`, then the
```
Comment (timhoffm) [inline lib/matplotlib/text.py]:
```suggestion
        If a `Text` instance is constructed with ``fontfamily=None``, then the
```

File: lib/matplotlib/text.py [hunk 3]
```diff
@@ -977,6 +977,11 @@ def set_fontfamily(self, fontname):
...
+        font is set to `matplotlib.rcParams['font.family']`, and the
```
Comment (timhoffm) [inline lib/matplotlib/text.py]:
```suggestion
        font is set to :rc:`font.family`, and the
```

Comment (Bharat123rox) [reply lib/matplotlib/text.py]: Done, hopefully nothing more comes up

**Comment by timhoffm** [comment]
Thanks, and congratulations on your first contribution to Matplotlib! Maybe see you back sometime.

**Comment by timhoffm** [comment]
@meeseeksdev backport to v3.1.0

**Comment by timhoffm** [comment]
@meeseeksdev backport to v3.1.x

**Comment by Bharat123rox** [comment]
Yes, you will definitely see me around...I like to think of this as "giving
back and enhancing" software that I use on a daily basis, so thanks for
reviewing it and hope to see you again soon!

**Pull request merged**